### PR TITLE
Name/Mac generation and libcontainer dep removal

### PR DESCRIPTION
**Pull request by nerdalert**
- Modified Mac address generation to match current standard
- Moved GenerateRandomName from libcontainer and removed the dependancy.
- Reduced entropy loop to 3 attempts.

Signed-off-by: Brent Salisbury <brent.salisbury@docker.com>

### CODE REVIEW

**Comment by dave-tucker** [comment]
LGTM

File: netutils/utils.go [hunk 1]
```diff
@@ -115,13 +116,34 @@ func GetIfaceAddr(name string) (net.Addr, []net.Addr, error) {
...
 }
+
+// GenerateRandomName returns a new name joined with a prefix.  This size
+// specified is used to truncate the randomly generated value
+func GenerateRandomName(prefix string, size int) (string, error) {
+	id := make([]byte, 32)
+	if _, err := io.ReadFull(rand.Reader, id); err != nil {
+		return "", err
+	}
+	return prefix + hex.EncodeToString(id)[:size], nil
```
Comment (mavenugo) [inline netutils/utils.go]: Shouldnt this be protected against `slice bounds out of range` for a size that is > 32 ?
Also, being an utility function, is there a technical reason to restrict this function to a 32 byte random value ?

File: netutils/utils_test.go [hunk 2]
```diff
@@ -174,3 +175,37 @@ func TestNetworkRange(t *testing.T) {
...
+		t.Fatalf("Expected 11 characters, instead received %d characters", len(name1))
+	}
+	name2, err := GenerateRandomName("veth", 7)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Fail if the random generated names equal one another
+	if name1 == name2 {
+		t.Fatalf("Expected differing values but received %s and %s", name1, name2)
+	}
```
Comment (mavenugo) [inline netutils/utils_test.go]: Needs more test-cases to cover negative conditions (such as `slice bounds out of range`)

File: netutils/utils.go [hunk 3]
```diff
@@ -115,13 +116,34 @@ func GetIfaceAddr(name string) (net.Addr, []net.Addr, error) {
...
+	// 1. Unicast: Set the least-significant bit to 0.
+	// 2. Address is locally administered: Set the second-least-significant bit (U/L) to 1.
+	// 3. As "small" as possible: The veth address has to be "smaller" than the bridge address.
+	hw[0] = 0x02
+	// The first 24 bits of the MAC represent the Organizationally Unique Identifier (OUI).
+	// Since this address is locally administered, we can do whatever we want as long as
+	// it doesn't conflict with other addresses.
+	hw[1] = 0x42
+	// Randomly generate the remaining 4 bytes (2^32)
+	buf := make([]byte, 4)
```
Comment (aboch) [inline netutils/utils.go]: You do not need buf.
Just have Read fill in hw[2:]

File: netutils/utils.go (hunk 1 above)
Comment (nerdalert) [inline netutils/utils.go]: I was just moving the function used in various places just to remove the libcontainer dependancy.
-github.com/docker/libcontainer/utils/utils.go
-github.com/docker/docker/pkg/common/randomid.go

**Comment by mrjana** [comment]
LGTM
